client/my-sites/email/email-management/home/email-plan-mailboxes-list.tsx

```diff
@@ -40,63 +50,104 @@ function EmailPlanMailboxesList( {
 		);
 	}
 
-	if ( ! mailboxes || mailboxes.length < 1 ) {
-		let missingMailboxesText = translate( 'No mailboxes' );
-
-		if (
-			isRecentlyRegistered( domain.registrationDate, 45 ) &&
-			getGSuiteSubscriptionStatus( domain ) === 'unknown'
-		) {
-			missingMailboxesText = translate(
-				'We are configuring your mailboxes. You will receive an email shortly when they are ready to use.'
-			);
-		}
-
+	if ( isGoogleConfiguring && configuringStateMode === 'message' ) {
 		return (
 			<MailboxListHeader accountType={ accountType }>
 				<MailboxListItem hasNoEmails>
-					<span>{ missingMailboxesText }</span>
+					<span>
+						{ translate(
+							'We are configuring your mailboxes. You will receive an email shortly when they are ready to use.'
+						) }
+					</span>
 				</MailboxListItem>
 			</MailboxListHeader>
```

Comment:
This block could be extracted to a small internal component (similar to `MailboxItemsEmpty`) as the same structure is used on both `isGoogleConfiguring` and `isNoMailboxes` conditionals. So, it could receive just the text as a prop.

Reply:
Yeah, nice suggestion. Thanks!

I refactored it, and now we have a reusable `MailboxContent` component.